Project/Gem/Source/Scripting/PayloadDespawnerComponent.cpp

```diff
@@ -53,10 +53,16 @@ namespace ROS2::Demo
                 if (IsObjectBox(entityId))
                 {
                     m_collidingBoxes.push_back(entityId);
+                    AZStd::string entityName;
+                    AZ::ComponentApplicationBus::BroadcastResult(entityName, &AZ::ComponentApplicationRequests::GetEntityName, entityId);
                 }
                 else if (IsObjectAMR(entityId))
                 {
-                    m_collidingAmr = entityId;
+                    if (!m_collidingAmr.IsValid())
+                    {
+                        m_collidingAmr = entityId;
+                        AZ_Printf("PayloadDespawnerComponent", "Detected AMR in despawner area.");
+                    }
```

Comment:
Is this log necessary? If not, validation here is also unnecessary as it is performed later when the entityId is used.